src/compose/app.ts

```diff
@@ -51,15 +51,26 @@ interface ChangingPair<T> {
 	target?: T;
 }
 
-export class App {
+export interface IApp {
```

Comment:
Why do we need this? Why not just use `App` in `common.ts`?

Reply:
That was the original intent, however type checks complain of missing public methods from the App class, such as `isTargetState`, `nextStepsForAppUpdate`, etc. I think this is due to the slight differences between current & target state, which are relevant here because we apply current state as target state in `doPurge` in common.ts.

Instead of declaring the `IApp` interface, I could specify the i/o types as:
```
export function safeAppClone(app: Partial<App>): Partial<App> {
``` 
or instantiate a new `App` instance for each `app` input into `safeAppClone`. Thoughts?

Reply:
Hmm, I think we need to improve the implementation of `doPurge`, I'm not sure why it needs to do the `safeStateClone` and eventually we need to clean up the way "intermediate" state is applied. In the meantime, I think instantiating an app in `safeAppClone` would solve the issue

`﻿``
export function safeAppClone(app: App): App {
   // ...
   return new App({appId: app.appId, /* add missing data here ... */}, true);
}
```

Reply:
Done with https://github.com/balena-os/balena-supervisor/pull/1842/commits/966d9574656bb393031ba64dd596f933eb972947!